social/strategies/django_strategy.py

```diff
@@ -63,7 +62,14 @@ def request_path(self):
 
     def request_port(self):
         """Port in use for this request"""
-        return self.request.META['SERVER_PORT']
+        try:  # django >= 1.9
+            return self.request.get_port()
+        except AttributeError:  # django < 1.9
+            host_parts = self.request.get_host().split(':')
+            try:
+                return host_parts[1]
+            except IndexError:
+                return None
```

Comment:
I should note - I have not authority over this repo whatsoever, I'm just a fellow user of PSA, trying to help out if I can :)

I'm worried about the behavior staying the same because things other than `python-saml` might be relying on `get_port()`.  Seems like the behavior in PSA shouldn't change dramatically (possibly returning `None` now) because of starting to use `USE_X_FORWARDED_HOST`.  So yeah it seems like falling back to old behavior when nothing else works might be more likely to result in the PR getting merged, but that's just my guess.